app/models/keyword.rb

```diff
@@ -0,0 +1,27 @@
+class Keyword < ActiveRecord::Base
+
+  has_many :questions
+
+  def self.select_21_link_in_random_order
+    Rails.cache.fetch("keyword_random", :expires_in => 5.seconds) do
+      select(:translit, :keyword)
+        .joins(questions: :user)
+        .uniq(:id)
+        .order("RAND()")
+        .first(21)
+    end
+  end
+
+  def with_questions_and_users
+    questions.includes(:user).includes(:answers)
```

Comment:
I don't really sure that second includes(:answers) would work. It should be tested and checked clean sql query like
```
questions.includes(:user).includes(:answers).to_sql
```
It should look like this
```
questions.includes(:user, :answers)
```
that how we could request multiple associations but it's NOT TESTED! maybe it would be needed any corrections